src/Vector3.h

```diff
@@ -289,6 +289,36 @@ class Vector3
 		);
 	}
 
+	/** Returns a copy of this vector moved by the specified offset on all axes. */
+	inline Vector3<T> added(Vector3<T> a_Offset) const
```

Comment:
I think `added{X,Y,Z}` are a nice convenience but I can't see why you would use `added` or `addedXZ` over just `operator+`.

Reply:
True, I thought I had a use case, but can't remember it now.

For the `XZ` version, we do have lots of code that scans only horizontal neighbors, not vertical, so they don't need to add the Y coord (and usually they don't even have one).

I'll get rid of the full `added`.